src/Main/Character/CharacterParses.js

```diff
@@ -0,0 +1,482 @@
+                            style={{ width: '100%' }} />
+                        </div>
+                      )}
+                      <h2 style={{ fontSize: '1.8em' }}>{this.props.region} - {this.props.realm}</h2>
+                      <h2 style={{ fontSize: '2.4em', margin: '10px 10px' }}>
+                        {this.props.name}
+                      </h2>
+                      {this.state.class && (
+                          <img
+                            src={`/specs/${this.state.class}-New.png`}
+                            alt="Pic"
+                            style={{ height: 50, position: 'absolute', right: 12, top: 10 }}
+                          />
+                        )}
+                    </div>
+                    <div className="col-md-4">
+                      Specs:
+                      {this.state.specs.map((elem, index) => 
+                        <div key={index} onClick={() => this.updateSpec(elem.replace(" ", ""))} className={this.state.activeSpec.includes(elem.replace(" ", "")) ? 'selected form-control' : 'form-control'}>
+                          <img src={this.iconPath(elem)} style={{ height: 18, marginRight: 10 }} alt="Icon" />
+                          {elem}
+                        </div>
+                      )}
+                    </div>
+                    
+                    <div className="col-md-4">
+                      Difficulties:
+                      {DIFFICULTIES.filter((elem) => { return elem; }).map((elem, index) => 
+                        <div key={index} onClick={() => this.updateDifficulty(elem)} className={ this.state.activeDifficulty.includes(elem) ? 'selected form-control' : 'form-control'}>
+                          {elem}
+                        </div>
+                      )}
+                    </div>
+                    <div className="col-md-4">
+                      Raid:
+                      <select className="form-control" onChange={e => this.setState({ activeZoneID: Number(e.target.value), activeEncounter: 0 }, () => { this.load().catch(e => { console.log('Error'); });})}>
+                        {Object.values(ZONES).reverse().map(elem =>
+                          <option key={elem.id} value={elem.id}>{elem.name}</option>
+                        )}
+                      </select>
+                      Boss:
+                      <select className="form-control" value={this.state.activeEncounter} onChange={e => this.setState({ activeEncounter: e.target.value })}>
+                        <option value={0} defaultValue>All bosses</option>
+                        {this.zoneBosses.map(e => 
+                          <option key={e.id} value={e.name}>{e.name}</option>
+                        )}
+                      </select>
+                      Metric:
+                      <select className="form-control" onChange={e => this.setState({ metric: e.target.value }, () => { this.load().catch(e => { console.log('Error'); });})}>
```

Comment:
handling errors like this will cause them to be muted and not appear in Sentry, which might make it hard to detect issues

Reply:
fixed by moving the `onChange` to functions that handle the state update & `this.load`